apps/userdashboard/assets/js/a4_candy_userdashboard/ModerationNotification.jsx

```diff
@@ -235,16 +236,18 @@ export default class ModerationNotification extends Component {
               <button className="btn btn--none" type="button" onClick={() => this.toggleIsPending()}><i className="fas fa-archive me-1" aria-hidden="true" />{unarchiveText}</button>
             </> /* eslint-disable-line react/jsx-closing-tag-location */}
         </div>
-        {this.state.moderatorStatementForm &&
+        {(this.state.moderatorStatementForm || this.state.isEditing) &&
           <ModerationStatementForm
-            onStatementSubmit={this.handleStatementSubmit}
+            onSubmit={this.handleStatementSubmit}
+            onEditSubmit={(payload, pk) => this.handleStatementEdit(payload, pk)}
             initialStatement={this.state.moderatorStatement}
+            editing={this.state.isEditing}
           />}
-        {this.state.moderatorStatement &&
+        {this.state.moderatorStatement && !this.state.isEditing &&
           <ModerationStatement
             statement={this.state.moderatorStatement}
             onDelete={this.handleStatementDelete}
-            onEdit={this.handleStatementEdit}
+            onEdit={() => this.setState({ isEditing: true })}
```

Comment:
we could also set showModeratorStatementForm to true here and then only use that flag to show form?